### DISCUSSION TITLE
Open Question: Showing information in the editor (or not?)

### DISCUSSION THREAD

**Discussion by agrignard** [gama-platform/gama#44] [General]
I know the information in the text editor can be a useful feature but the way it works is in my opinion strange.

1. As soon as you start to write a line unless your line is finished and compile the information disapears for a while and the hole text on the top is moving  which in my opinion make it very difficult to concentrate as shown below:

![Untitled mov](https://github.com/gama-platform/gama/assets/3928502/2a3ca85f-d3e8-462d-a040-9568d795ab45)

2. Having some text in your text editor than cannot be copy/paste and take one "pseudo line" might be also conter intuitive
<img width="785" alt="Screenshot 2024-02-20 at 23 46 15" src="https://github.com/gama-platform/gama/assets/3928502/8b9d9832-7345-407a-8e62-f0afa02f94c9">

3. Hard to differentiate comment form information too

I know it's a  question fo choice and can be set in the preferences however most of the user will keep the default settings and might be annoyed by this. 

What about settings those parameters to false by default ?

**Comment by agrignard** [comment]
Trade off proposal: What about keeping only the thumbnails on the left side of the screen  ? By default the information/warning are still here if needed but does not impact the editor by adding new "lines" in your code and make it moves (like in may other editors actually)? 

Instead of having both thumbnails for the same information
<img width="689" alt="Screenshot 2024-02-21 at 10 04 36" src="https://github.com/gama-platform/gama/assets/3928502/381f8b23-f81a-4519-9151-184bf045b663">

**Reply by lesquoyb** [reply]
I agree that having the thumbnail on the line and above is a bit redundant and adds complexity for nothing.
Though that "trade off" is actually the previous mode, but maybe we can let the user chose between nothing, thumbnails on the left and thumbnail above + description. I'd also be in favor of having that setting available directly in one of the top menus as it's the case in most IDEs
